test/testdata/namer/constant_redefinition/nested_module_then_nonexistent_name.rb

```diff
@@ -0,0 +1,9 @@
+# typed: true
+
+module A::B::C
+  def foo; end
+end
+
+  A::B = A::D
+# ^^^^^^^^^^^ error: Redefining constant `B`
+       # ^^^^ error: Unable to resolve constant `D`
```

Comment:
What do you think about adding something to the output of `ast::ConstantLit::showRaw` to add a field like `symbol-kind` to pretty print the symbol's kind?

Because I believe that this change also fixes the bug where we were symbolizing the trees wrong, and it'd be nice to not regress on that.

Reply:
Discussion elsewhere says we can delay this for a followup.